src/Microsoft.VisualStudio.Editors/DesignerFramework/AccessModifierComboBox.vb

```diff
@@ -405,12 +406,23 @@ Namespace Microsoft.VisualStudio.Editors.DesignerFramework
         Private Function TryGetCustomToolPropertyValue(ByRef value As String) As Boolean
             value = Nothing
 
-            Dim customToolProperty As EnvDTE.Property = DTEUtils.GetProjectItemProperty(_projectItem, DTEUtils.PROJECTPROPERTY_CUSTOMTOOL)
-            If customToolProperty IsNot Nothing Then
-                Dim customToolValue As String = TryCast(customToolProperty.Value, String)
-                value = customToolValue
-                Return True
-            End If
+            Try
+                Dim customToolProperty As EnvDTE.Property = DTEUtils.GetProjectItemProperty(_resxFileProjectItem, DTEUtils.PROJECTPROPERTY_CUSTOMTOOL)
+                If customToolProperty IsNot Nothing Then
+                    Dim customToolValue As String = TryCast(customToolProperty.Value, String)
+                    value = customToolValue
+                    _previousCustomToolValue = customToolValue
+                    Return True
+                End If
+            Catch ex As KeyNotFoundException
```

Comment:
I assume you are only trying to catch `KeyNotFoundException` from the `DTEUtils.GetProjectItemProperty` call. Perhaps the `try`/`catch` block should only include that call.

Reply:
Yeah. That would make more explicit about the code that might throw the exception.